internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -4,26 +4,111 @@ import (
+     |                          |                                  |<-----------------------|
+     |                          |                                  |                        |
+     |                          |                                  | Login Request          |
+     |                          |                                  |----------------------->|
+     |                          |                                  |                        |
+     |                          |                                  |         Login Response |
+     |                          |                                  |<-----------------------|
+     |                          |                                  |                        |
+     |                          |         Login success or failure |                        |
+     |                          |<---------------------------------|                        |
+     |                          |                                  |                        |
+     |    Login Response (Fake) |                                  |                        |
+     |<-------------------------|                                  |                        |
+     |                          |                                  |                        |
+
+	Note: The above diagram was created using https://textart.io/sequence and the
+	following source:
+
+	object Client Secretless Driver MsSQL
+	Client->Secretless: Prelogin Request
+	Secretless->Driver: Connect(context)
+	note right of Secretless: Context contains channels \n for intercepting data
+	Driver->MsSQL: Prelogin Request
+	MsSQL->Driver: Prelogin Response
+	MsSQL->Secretless: Intercept Prelogin Response
+	Secretless->Secretless: Modify Prelogin Response
+	Secretless->Client: Prelogin Response
+	Driver->MsSQL:Handshake Request
+	MsSQL->Driver: Handshake Response
+	Driver->MsSQL: Login Request
+	MsSQL->Driver: Login Response
+	Driver->Secretless: Login success or failure
+	Secretless->Client: Login Response (Premade)
+ */
+
 // SingleUseConnector is used to create an authenticated connection to an MSSQL target
 type SingleUseConnector struct {
 	backendConn net.Conn
 	clientConn  net.Conn
 	logger      log.Logger
 }
 
+const bufferSize uint16 = 4086
+
 // Connect implements the tcp.Connector func signature
 //
 // It is the main method of the SingleUseConnector. It:
 //   1. Constructs connection details from the provided credentials map.
-//   2. Dials the backend using credentials.
-//   3. Runs through the connection phase steps to authenticate.
+//   2. Reads the client PreLogin request
+//   3. Adds channels to the context that can intercept data from the driver
+//   4. Initiates authentication and connection to MsSQL through the third-party driver
+//   6. Intercepts PreLogin response from the driver
+//	 5. Customizes the PreLogin response to meet Secretless standards
```

Comment:
I think 5 and 6 are swapped

Reply:
good catch, meant to change the numbers there